R/http_methods.R

```diff
@@ -0,0 +1,185 @@
+#' @include utils.R
+#' @importFrom assertthat assert_that is.scalar
+NULL
+
+#' Chrome Debugging Protocol HTTP methods
+#'
+#' When Chromium/Chrome is started in debugging mode, several HTTP endpoints
+#' are available. Using these endpoint, one can retrieve information from
+#' Chromium/Chrome or send commands, see
+#' <https://chromedevtools.github.io/devtools-protocol/#endpoints>.
+#'
+#' @param host Character scalar, the host name of the application.
+#' @param port The remote debugging port (a numeric or a character scalar).
+#' @param secure A logical indicating whether a secure protocol shall be used.
+#' @param local Logical scalar, indicating whether the local version of the
+#'     protocol (embedded in `crrri`) must be used or the protocol must be
+#'     fetched _remotely_.
+#' @param url URL to open in a new tab.
+#' @param target_id Target (or tab) identifier.
+#' @return `fetch_version()`, `list_targets()`, `fetch_protocols()` and
+#'   `new_tab()` return a list. `activate_target()` and `close_target()` returns
+#'    a logical: `TRUE` is returned when the command succeeds, `FALSE` is
+#'    returned if a wrong `target_id` is provided.
+#' @name http-methods
+#' @keywords internal
+#' @examples
+#' \donttest{
+#' chrome <- Chrome$new()
+#'
+#' # fetch information about headless Chrome
+#' fetch_version()
+#' # fetch the Chromium/Chrome protocol
+#' protocol <- fetch_protocol()
+#'
+#' # get the list of opened tabs
+#' list_targets()
+#' # open an url in a new tab
+#' new_target <- new_tab(url = "http://r-project.org")
+#' # the new tab is referenced in the list of opened tabs
+#' list_targets()
+#' # inspect a target in a web browser
+#' if(interactive()) inspect_target(target_id = new_target$id)
+#' # close the previous created tab
+#' close_target(target_id = new_target$id)
+#'
+#' chrome$close()
+#' }
+NULL
+
+#' @rdname http-methods
+#' @export
+fetch_version <- function(host = "localhost", port = 9222, secure = FALSE) {
+  fetch_json(host, port, secure, "version")
+}
+
+#' @rdname http-methods
+#' @export
+list_targets <- function(host = "localhost", port = 9222, secure = FALSE) {
```

Comment:
Naming things is always painful.
My first idea was to use `fetch_*` where `*` was the last part of the endpoint path. 
This had lead to `fetch_version()`, `fetch_list()`, `fetch_protocol()` and `fetch_new()`. I found `fetch_protocol()` and `fetch_version()` meaningful but struggled with `fetch_new()` and `fetch_list()`.
Using `fetch_targets()` instead of `list_targets()` makes sense... but we would lose the reference to the [`list` endpoint](https://chromedevtools.github.io/devtools-protocol/#get-json-or-jsonlist).
My main problem was with `fetch_new()`: I picked `new_tab()` in order to have a reference to the `new` endpoint.
Finding a good naming convention for endpoints is difficult. I do not mind a lot about these things: it is ok for me to use `fetch_targets` and `get_browser_endpoint`.

Reply:
Yes, I think there is no good name for now - those are not better at least. Thanks!